icu4c/source/common/unicode/umachine.h

```diff
@@ -377,9 +377,17 @@ typedef int8_t UBool;
 #endif
 
 #if defined(U_COMBINED_IMPLEMENTATION) || defined(U_COMMON_IMPLEMENTATION) || \
-        defined(U_I18N_IMPLEMENTATION) || defined(U_IO_IMPLEMENTATION)
+        defined(U_I18N_IMPLEMENTATION) || defined(U_IO_IMPLEMENTATION) || \
+        defined(U_TOOLUTIL_IMPLEMENTATION)
     // Inside the ICU library code, never configurable.
     typedef char16_t UChar;
+#elif defined(T_CTEST_IMPLEMENTATION)
+    // internally to ctestfw, we want to use char16_t in C++ and uint_16 in C
```

Comment:
Why do we need to default to uint16_t in C? We are on C11 now, and char16_t is part of the standard.
https://en.cppreference.com/w/c/string/multibyte/char16_t

Reply:
That wasn't true when I originally made this fix in Apple ICU, but that was before we changed minimum C versions.  Let me try it again now...

Reply:
Well, **THIS** is bad...

I just figured out the hard way that Apple platforms don't actually support C11.  Or maybe they do partially, but they don't have the `<uchar.h>` header, where `char16_t` is defined.  From looking at the existing bug reports about this, I'm not getting the impression anyone's going to fix it on Apple platforms any time soon.  So I either need to keep this code as is or simplify it for other platforms and come up with a workaround for Apple platforms...